etc/http/openui5/controller/Canvas.controller.js

```diff
@@ -79,94 +161,120 @@ sap.ui.define([
-            });
+         var oContent = sap.ui.xmlview({
+            viewName : "sap.ui.jsroot.view." + panel_name,
+            layoutData: oLd
+         });
 
-            split.insertContentArea(oContent, 0);
+         split.insertContentArea(oContent, 0);
 
-            var ctrl = oContent.getController();
-            p.SelectObjectPainter = ctrl.onObjectSelect.bind(ctrl, p);
+         return oContent.getController(); // return controller of new panel
+		},
 
-            ctrl.onObjectSelect(p,p);
+	   ShowCanvasStatus : function (text1,text2,text3,text4) {
+	      var model = this.getView().getModel();
+	      model.setProperty("/StatusLbl1", text1);
+	      model.setProperty("/StatusLbl2", text2);
+	      model.setProperty("/StatusLbl3", text3);
+	      model.setProperty("/StatusLbl4", text4);
+      },
 
-         } else {
-            split.removeContentArea(split.getContentAreas()[0]);
-            delete p.SelectObjectPainter;
-         }
+		isStatusShown : function() {
+		   return this._Page.getShowFooter();
+		},
+
+		toggleShowStatus : function() {
+		   var new_state = !this.isStatusShown();
+
+         this._Page.setShowFooter(new_state);
+         this.getView().getModel().setProperty("/StatusIcon", new_state ? "sap-icon://accept" : "");
 		},
 
 		onViewMenuAction : function (oEvent) {
-         var p = this.getCanvasPainter(true);
-         if (!p) return;
 
          var item = oEvent.getParameter("item"),
              name = item.getText();
 
+         if (name=="Editor") return this.toggleGedEditor();
+         if (name=="Event statusbar") return this.toggleShowStatus();
+
+
+         var p = this.getCanvasPainter(true);
+         if (!p) return;
+
          var new_state = !item.getIcon();
 
          switch (name) {
-            case "Editor":
-               this.toggleGedEditor();
-               return;
             case "Toolbar":
                this._Page.setShowSubHeader(new_state)
                break;
-            case "Event statusbar":
-               this._Page.setShowFooter(new_state);
-               if (new_state) {
-                  p.ShowStatus = this.ShowCanvasStatus.bind(this);
-               } else {
-                  delete p.ShowStatus;
-               }
+            case "Tooltip info":
+               p.SetTooltipAllowed(new_state);
                break;
-            case "Tooltip info": p.SetTooltipAllowed(new_state); break;
             default: return;
          }
-
          item.setIcon(new_state ? "sap-icon://accept" : "");
 
          // MessageToast.show("Action triggered on item: " + name);
+		},
+
+		onToolsMenuAction : function(oEvent) {
+         var item = oEvent.getParameter("item"),
+             name = item.getText();
+
+         if (name != "Fit panel") return;
+
+         var curr = this.getView().getModel().getProperty("/LeftArea");
+
+         this.showLeftArea(curr == "FitPanel" ? "" : "FitPanel");
+		},
+
+		showMessage : function(msg) {
+		   MessageToast.show(msg);
```

Comment:
Could you adjust the code indentation inside `onToolsMenuAction : function(oEvent)`?